includes/api/class-wc-rest-coupons-controller.php

```diff
@@ -502,7 +502,7 @@ public function get_item_schema() {
 				),
 				'meta_data' => array(
 					'description' => __( 'Order meta data.', 'woocommerce' ),
-					'type'        => 'array',
```

Comment:
@mikejolley Sorry, but i can't test it in moment. But i think that when a list of items will given back that is a normal numeric array. WordPress 4.7 uses the `wp is_numeric_array()` function to check the property.